backend/app/quiz_creation/conjugation_quiz.py

```diff
@@ -0,0 +1,189 @@
+import random
+from app.analysis.parts_of_speech import get_parts_of_speech_tags, CONTRACTIONS
+from app.quiz_creation.verb_conjugation import (
+    verb_tenses,
+    verb_tense,
+    verb_tenses_keys,
+    verb_infinitive,
+)
+
+
+def get_sentence_options(word):
+    """
+    Creates the answer choices for a sentence in the quiz game
+    :param word: A verb (string) that is the answer to the question
+    :return: A list of 4 words to serve as the answer choices
+    """
+    options = [word]
+    word_is_capitalized = word[0].isupper()
+    # Check if tense is found in the tenses dictionary
+    tenses = verb_tenses.get(verb_infinitive(word), '')
+    if tenses:
+        # Determine the number of discrete tenses of the verb
+        discrete_tenses = []
+        for tense in tenses:
+            if tense != '' and tense != word and tense not in discrete_tenses:
+                discrete_tenses.append(tense)
+
+        if len(discrete_tenses) >= 3:
+            options += random.sample(discrete_tenses, 3)
+        else:
+            # If the verb does not have at least 3 discrete verb tenses, add the same tense of
+            # another verb. ex: word = walked, other_options = baked, raked, etc.
+            verb_form = verb_tense(word)
+            key = verb_tenses_keys[verb_form]
+            verbs_of_same_tense = [
+                verb_tenses[verb][key] for verb in verb_tenses if verb_tenses[verb][key] != ''
+            ]
+            options += discrete_tenses
+            options += random.sample(verbs_of_same_tense, 3 - len(discrete_tenses))
+    else:
+        # If the verb is not in the verb_tenses dictionary, get three random verb infinitives
+        # (keys) from the dictionary
+        options += random.sample(list(verb_tenses), 3)
+
+    # Capitalize other options if the original word was capitalized
+    if word_is_capitalized:
+        for i in range(4):
+            options[i] = options[i].capitalize()
+
+    random.shuffle(options)
+
+    # Capitalize the options if verb is at the beginning of the sentence
+    if word[0].isupper():
+        return [w.capitalize() for w in options]
+
+    return options
+
+
+def process_text(text):
+    """
+    Removes contractions and replaces direct quotation marks with their python escape sequence
+    equivalents
+    :param text: A body of text as a string
+    :return: text: A body of text as a string with no contractions
+    """
+
+    # remove_contractions is buggy
+    # processed = remove_contractions(text)
+    # need to test on input text from frontend
+    processed = text.replace('"', '\"')
+    processed = processed.replace('’', "\'")
+
+    return processed
+
+
+
+def get_quiz_sentences(text):
```

Comment:
This function needs a little bit of a refactor for clarity. Right now, it consumes tagged words left to right, does some processing on those words as it goes, and then when we find that the next token is a sentence-ending punctuation mark, we do a big inner loop of adding the right blank and the correct answer to the data structure.

While this works just fine, it means to understand the function, you have to read the whole thing linearly, and if something breaks in a particular part it'll be difficult to debug.

Here's my suggestion for a refactor: _first_ tokenize the whole text into sentences. Then per sentence run the inner loops from the existing function here, but broken out into helpers. It'll look something like this:

```
sentences = sentence_tokenize(text)  # NLTK has a sentence tokenizer that we coulduse
possible_quiz_sentence_dicts = []  # these might not pass validation bc they don't have answers

for sentence in sentences:
    current_sentence = {
        'sentence': [],
        'options': [],
        'answer': None
    }
    sentence_text = process_text(sentence)
    pos_tags = get_parts_of_speech_tags(sentence_text)
    for i, [word, pos] in enumerate(pos_tags):
        # factor this whole chunk out into a helper function
    possible_quiz_sentence_dicts.append(current_sentence)

# once we're done preprocessing the sentences, iterate over the possible sentences....
for sentence_dict in possible_quiz_sentence_dicts:
    # now the whole chunk of logic under
    #        if pos == '.' or i == len(pos_tags) - 1:
    # called out...
    select_verb_and_make_fillin(sentence_dict)
    if has_valid_answer(sentence_dict):
        quiz_sentences.append(current_sentence)
```

This way the purpose of each 20-30 line chunk of code is clearly named and factored out, can be _tested_ individually, and this big function will become a lot clearer.